src/CMakeLists.txt

```diff
@@ -300,6 +300,7 @@ endif()
 # tcmalloc heap profiler
 set(heap_profiler_files ${TCMALLOC_srcs})
 add_library(heap_profiler_objs OBJECT ${heap_profiler_files})
+add_dependencies(heap_profiler_objs Boost::boost)
```

Comment:
> for example, to preprocess CMakeLists.txt using cmake and add add_dependencies(${THE_TARGET} Boost::boost) for every target.

we might consider something [hacky like this](https://cmake.org/pipermail/cmake/2014-August/058418.html), which overrides `add_library()` and `add_executable()` to inject a global dependency